transport_analysis/analysis/velocityautocorr.py

```diff
@@ -5,111 +5,200 @@
+
+        self.dim_type = self.dim_type.lower()
+
+        try:
+            self._dim = keys[self.dim_type]
+        except KeyError:
+            raise ValueError(
+                "invalid dim_type: {} specified, please specify one of xyz, "
+                "xy, xz, yz, x, y, z".format(self.dim_type)
+            )
+
+        self.dim_fac = len(self._dim)
 
     def _single_frame(self):
-        """Calculate data from a single frame of trajectory"""
+        """Constructs array of velocities for VACF calculation."""
         # This runs once for each frame of the trajectory
-        # It can contain the main analysis method, or just collect data
-        # so that analysis can be done over the aggregate data
-        # in _conclude.
 
         # The trajectory positions update automatically
-        negative = self.atomgroup.positions < 0
         # You can access the frame number using self._frame_index
-        self.results.is_negative[self._frame_index] = negative.any(axis=1)
 
+        # trajectory must have velocity information
+        if not self._ts.has_velocities:
+            raise NoDataError(
+                "VACF computation requires velocities " "in the trajectory"
+            )
+
+        # set shape of velocity array
+        self._velocity_array[self._frame_index] = self.atomgroup.velocities[
+            :, self._dim
+        ]
+
+    # Results will be in units of (angstroms / ps)^2
     def _conclude(self):
         """Calculate the final results of the analysis"""
         # This is an optional method that runs after
         # _single_frame loops over the trajectory.
         # It is useful for calculating the final results
         # of the analysis.
-        # For example, below we determine the
-        # which atoms always have negative coordinates.
-        self.results.always_negative = self.results.is_negative.all(axis=0)
-        always_negative_atoms = self.atomgroup[self.results.always_negative]
-        self.results.always_negative_atoms = always_negative_atoms
-        self.results.always_negative_atom_names = always_negative_atoms.names
-
-        # results don't have to be arrays -- they can be any value, e.g. floats
-        self.results.n_negative_atoms = self.results.is_negative.sum(axis=1)
-        self.results.mean_negative_atoms = self.results.n_negative_atoms.mean()
+        if self.fft:
+            self._conclude_fft()
+        else:
+            self._conclude_simple()
+
+    def _conclude_fft(self):  # with FFT, np.float64 bit prescision required.
+        r"""Calculates the VACF via the FCA fast correlation algorithm."""
+        try:
+            import tidynamics
+        except ImportError:
+            raise ImportError(
+                """ERROR --- tidynamics was not found!
+
+                tidynamics is required to compute an FFT based VACF (default)
+
+                try installing it using pip eg:
+
+                    pip install tidynamics
+
+                or set fft=False"""
```

Comment:
I fear that because you are using a triple quote the white space will be rendered in the error message and make it less readable. Could you check that? If so, you could replace these with a set of quotes in parentheses. e.g.

```python
long_quote = (
    " text text text",
    " text text text",
    " text text text",
    " text text text"
)
```

I think thats the correct syntax, you may not even need the commas

Reply:
So I've checked it on my machine (Fedora Linux 38) and it shows up fine in both the terminal and Jupyter Lab. Feel free to check it on yours as well and let me know if you see anything strange!

Reply:
Got it, good to know. I am curious if there is any PEP guidance on triple quotes vs parenthetical single quotes, does anyone else have an opinion here?

Reply:
Should be fine with triple quotes but see my  opinion on forcing tidynamics to be installed.